packages/server/parseFormBody.ts

```diff
@@ -0,0 +1,80 @@
+import {HttpResponse, HttpRequest} from 'uWebSockets.js'
+import Busboy from 'busboy'
+import {Readable} from 'stream'
+import {OutgoingMessage} from '@mattkrick/graphql-trebuchet-client'
+import {APP_MAX_AVATAR_FILE_SIZE} from 'parabol-client/utils/constants'
+
+const bodyStream = (res: HttpResponse) => {
+  const stream = new Readable()
+  stream._read = () => 'no-op'
+  res.onData((ab, isLast) => {
+    const curBuf = Buffer.from(ab)
+    stream.push(new Uint8Array(curBuf.slice(curBuf.byteOffset, curBuf.byteLength)))
+    if (isLast) stream.push(null)
+  })
+  return stream
+}
+
+const reqHeaders = (req: HttpRequest) => {
+  const headers = {} as {[key: string]: string}
+  req.forEach((k, v) => (headers[k] = v))
+  return headers
+}
+
+const parseFile = (fileStream, contentType) => {
+  return new Promise((resolve, reject) => {
+    let buffer: Buffer
+    fileStream.on('data', (curBuf) => {
+      buffer = buffer ? Buffer.concat([buffer, curBuf]) : Buffer.concat([curBuf])
```

Comment:
this is the real steak & potatoes of the PR! this piece concerns me a bit because we already copied the buffer once as it came in & here we're copying it again.
i think we can entirely get rid of busboy & use uWS. Alex just released a version with multipart handling (https://github.com/uNetworking/uWebSockets.js/releases/tag/v18.7.0). 

Note that upgrading has a breaking change so you'll have to refactor `handleOpen` to use an upgrade event. He has examples included in the repo (https://github.com/uNetworking/uWebSockets.js/releases/tag/v18.0.0). You can break that into its own PR if you like

Reply:
upgraded to v18.10.0 bc [the interface of `MultipartField[]` changed from a map to an object data structure](https://github.com/uNetworking/uWebSockets.js/issues/375), and it is easier to work with objects in typescript.

Reply:
also forked the repo bc [the type definition file in the binaries](https://github.com/uNetworking/uWebSockets.js/tree/v18.10.0) is out-of-date than [the one on master](https://github.com/uNetworking/uWebSockets.js/blob/master/docs/index.d.ts), and the out of date one doesn't include the new api for `getParts`.